rcon/commands.py

```diff
@@ -138,7 +138,8 @@ def __init__(self, config, auto_retry=1, max_open=None, max_idle=None):
     @contextmanager
     def with_connection(self) -> HLLConnection:
         logger.debug("Waiting to acquire lock %s", threading.get_ident())
-        self.mu.acquire()
+        if not self.mu.acquire(timeout=30):
```

Comment:
You could make it more agressive. If the nb of connection in the pool matches the number of gunicorn threads per worker then a thread should never have to wait to get a connection in theory

Reply:
That would, however, only work for the backend. The pool is the same for background tasks (like automod, routines, log_event and so on). In these case, the thread count might be different from the number of connections in the pool. And I would love to have the pool Independent from configuration in gunicorn 😅